x-pack/plugins/alerting/server/connector_adapters/types.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ObjectType } from '@kbn/config-schema';
+import type {
+  RuleActionParams as GenericRuleActionParams,
+  RuleTypeParams,
+  SanitizedRule,
+} from '../../common';
+import { CombinedSummarizedAlerts } from '../types';
+
+type ActionTypeParams = Record<string, unknown>;
+
+type Rule<RuleParams extends RuleTypeParams = RuleTypeParams> = Pick<
+  SanitizedRule<RuleParams>,
+  'id' | 'name' | 'tags'
```

Comment:
For the MVP this should be enough. If a connector adapter needs more attributes from the rule we can extend it.